### PR TITLE
[REVIEW] renumber vertex ids

### LINKED ISSUE #592: [BUG] Accept user's notation when passing vertices ID to analytics
Now that we have auto renumbering (when loading a graph) and unrenumbering (when returning an analytics result). We should also update analytics that accepts vertex IDs so that they match the user's scope and not our own internal representation.

- Update Jaccard to accept unremembered IDs for arbitrary pairs of vertices specified by the user.
- Update Personalized PageRank to accept unremembered IDs for the personalization parameter
- Pagerank guess should be a key-value DF of size V like in nstart for Katz.
- Double-check the API to make sure we are not assuming renumbered input anywhere.
- Katz
- Subgraph extraction
- sssp
- BFS
- K-core
- degrees functions

### PR DESCRIPTION
**Pull request by Iroy30**
This PR takes user notations into consideration when vertices are passed as arguments for analytics. If the graph is renumbered, passed vertices are also auto-renumbered according to the renumbering map used for the graph.
Changed first and second cudf.Series (to provide vertex pairs for computation of coefficient) in jaccard, wjaccard, overlap and woverlap to  a cudf.DataFrame input since it is more intuitive.
closes #592

### CODE REVIEW

**Comment by afender** [comment]
These algorithms also accept V IDs and should be upgraded:
- Katz
- Subgraph extraction
- sssp
- BFS
- K-core
- degrees functions

**Comment by Iroy30** [comment]
rerun tests

File: python/cugraph/link_prediction/jaccard.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 import cudf
 
 
-def jaccard(input_graph, first=None, second=None):
+def jaccard(input_graph, vertex_pair=None):
```
Comment (afender) [inline python/cugraph/link_prediction/jaccard.py]: Please add a note for this API change in the description of the PR

File: python/cugraph/link_prediction/jaccard.py [hunk 2]
```diff
@@ -36,12 +36,8 @@ def jaccard(input_graph, first=None, second=None):
...
         directed edge in both direction. The adjacency list will be computed if
         not already present.
-    first : cudf.Series
-        Specifies the first vertices of each pair of vertices to compute for,
-        must be specified along with second.
-    second : cudf.Series
-        Specifies the second vertices of each pair of vertices to compute for,
-        must be specified along with first.
+    vertex_pair : cudf.DataFrame
+        Specifies the pair of vertices to compute for
```
Comment (afender) [inline python/cugraph/link_prediction/jaccard.py]: Doc needs improvements (see slides 13 to 30 from RAPIDS review and doc talk)

File: python/cugraph/link_prediction/overlap.py [hunk 3]
```diff
@@ -34,12 +34,8 @@ def overlap(input_graph, first=None, second=None):
...
         as an edge list (edge weights are not used for this algorithm). The
         adjacency list will be computed if not already present.
-    first : cudf.Series
-        Specifies the first vertices of each pair of vertices to compute for,
-        must be specified along with second.
-    second : cudf.Series
-        Specifies the second vertices of each pair of vertices to compute for,
-        must be specified along with first.
+    vertex_pair : cudf.DataFrame
+        Specifies the pair of vertices to compute for
```
Comment (afender) [inline python/cugraph/link_prediction/overlap.py]: Doc should be improved

File: python/cugraph/link_prediction/jaccard.py (hunk 1 above)
Comment (Iroy30) [reply python/cugraph/link_prediction/jaccard.py]: added

**Comment by afender** [comment]
Let's update the notebook for the API change

**Comment by Iroy30** [comment]
rerun tests

**Comment by Iroy30** [comment]
rerun tests

**Comment by Iroy30** [comment]
Corresponding vertex_pair api update in rapidsai/notebooks PR 292

**Comment by afender** [comment]
Notebook's PR link for github : https://github.com/rapidsai/notebooks/pull/292

**Pull request merged**